### ISSUE TITLE
Different spectra from SHExpandGLQ() -> spectrum() in version 4.9.1 and 4.10

### ISSUE THREAD

**Issue by oanegros** [SHTOOLS/SHTOOLS#376]
Hi, 
I was running GLQ-sampled expansion of my data and get very different power spectra when running the following block with pyshtools 4.9.1 and 4.10:
```
projection =  np.load(fname)
zero, w = pysh.expand.SHGLQ(251)
glq = pysh.expand.SHExpandGLQ(projection, zero=zero, w=w)
spec = pysh.spectralanalysis.spectrum(glq, unit="per_dlogl", base=2)
```
Where I get the following screenshot with 4.9.1:
<img width="896" alt="Screenshot 2023-03-07 at 12 56 28" src="https://user-images.githubusercontent.com/42999309/223416752-9fa67d84-d79e-41f8-85a6-c0601863dad8.png">
and the following with 4.10:
<img width="883" alt="Screenshot 2023-03-07 at 12 57 55" src="https://user-images.githubusercontent.com/42999309/223416838-a7e56c1c-1bae-490b-9c7a-2adfff66de85.png">
Made with the underlying data and plotting functions.

With `pysh.expand.MakeGridGLQ(coeffs, zero)` i am able to regenerate the original data from both expansions.

I couldnt find any references to this in #340, was this a consequence of the backend change? Is the 4.10 version the correct spectrum? And if so, is there a way to force the 4.9.x version to make this?
```
f, ax = plt.subplots(figsize=(9, 4))
plt.plot(np.arange(0,251),spec)
plt.yscale('log',base=2) 
plt.xscale('log',base=2) 
plt.show()
```
![img_of_array](https://user-images.githubusercontent.com/42999309/223416940-18b8c4e7-b787-4f33-98f2-77e43ce82191.png)
[testprojection.csv](https://github.com/SHTOOLS/SHTOOLS/files/10909246/testprojection.csv)

**System information**

**Comment by oanegros** [comment]
When running plot_spectrum with default normalization:
```
zero, w = pysh.expand.SHGLQ(251)
glq = pysh.expand.SHExpandGLQ(projection, zero=zero, w=w)
clm = pysh.SHCoeffs.from_array(glq)
clm.plot_spectrum()
plt.show()
```
I still get this effect:
v4.9.1
<img width="708" alt="Screenshot 2023-03-07 at 15 12 39" src="https://user-images.githubusercontent.com/42999309/223447492-78b6d153-a8d5-47ec-baad-a6608298886f.png">

v4.10
<img width="706" alt="Screenshot 2023-03-07 at 15 12 33" src="https://user-images.githubusercontent.com/42999309/223447511-a9db5936-1b80-4737-8543-e689b4773c34.png">

**Comment by MarkWieczorek** [comment]
So which is right? I would assume the v4.10 spectra is the right one. Perhaps you could try reinstalling v4.9.1?

Also, it would be good to verify that the coefficients "clm" are the same using both versions. Given how different the spectra are, it should be easy to just make sure that the magnitudes are correct for a couple coefficients.

**Comment by oanegros** [comment]
Hi,
I figured out part of what i was doing wrong to cause this behaviour, I was using `zero, w = pysh.expand.SHGLQ(251)` for a 251x502 map, that needed `zero, w = pysh.expand.SHGLQ(250)`.
It seems that my major issue of the weird maps is fixed with this, I was sampling slightly wrong also upstream to create the map (the 251 should have been correct, with a resulting 252x503 map).

However, for the behavior of the library, it is still weird that the two versions have different behavior. It seems to me that the newer 4.10 version inferred the right weights/nodes for the map without saying so, but i dont know for sure?
Also, the 4.9 and 4.10 versions do not pass np.array_equiv(clm1, clm2) with `zero, w = pysh.expand.SHGLQ(250)`.
with this crude quantification:
```
diffs = []
for i in range(glq.shape[0]):
    for j in range(glq.shape[1]):
        for k in range(glq.shape[2]):
            diffs.append(np.abs(glq[i][j][k] - v10version[i][j][k]))
print(sum(diffs))            
```
This still yields a difference of 109.06830434422352 after setting the correct zero, w (it was 20552.55470896045 with the wrong 251 L)  
Do you know where this change in behavior comes from, and is it defined somewhere?
Thanks

**Comment by MarkWieczorek** [comment]
I'm not really sure what to say. If you could give me a simple script that reproduces this problem, then I'll look into it. I looked at the code, and it doesn't look like the routine `pysh.expand.SHGLQ` has been changed since Jan 2021, and even then, that change was trivial. 

For the differences you show, its hard for me to guess, because I don't know what the dataset looks like and what the amplitude of the coefficients are. What I would do is plot the power spectrum of the two sets of coefficients, and the power spectrum of the difference.

**Comment by MarkWieczorek** [comment]
But there aren't any macOS ARM conda packages for pyshtools.... Could you let me know how you are installing pyshtools? I can reproduce the results for v4.10, so I am guessing that there was something wrong with your 4.9.1 install.

**Comment by oanegros** [comment]
Yes, so this was using rosetta (the Mac M1  ARM to x86 translation layer), which was specified with the `CONDA_SUBDIR=osx-64` flag in my install command: `CONDA_SUBDIR=osx-64 mamba create --name test_shtools9_1 "pyshtools=4.9.1" -c conda-forge`

**Comment by MarkWieczorek** [comment]
Could you let me know if you ever resolved this problem? We just put out a new release today (4.13.1), so you might want to try updating.

**Comment by MarkWieczorek** [comment]
Could you let me know if this is a problem with the latest release (4.14.1)?

**Comment by aafkegros** [comment]
Sorry for not responding. I believe this was caused by a wrongly installed package in my base conda that had weird downstream effects on the virtual environment. It was fixed, I forgot to respond here, thanks for looking into it!
